tests/preprocessing/test_triplegs.py

```diff
@@ -43,6 +43,24 @@ def test_generate_trips(self):
         # test if generated trips are equal
         pd.testing.assert_frame_equal(trips_loaded, trips)
         
+    def test_generate_trips_missing_link(self):
+        """Test nan is assigned for missing link between spts and trips, and tpls and trips."""
+        # create trips from geolife (based on positionfixes)
+        pfs = ti.io.dataset_reader.read_geolife(os.path.join('tests', 'data', 'geolife_long'))
+        pfs, stps = pfs.as_positionfixes.generate_staypoints(method='sliding', 
+                                                             dist_threshold=25,
+                                                             time_threshold=5 * 60)
+        stps = stps.as_staypoints.create_activity_flag()
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(stps)
+        
+        # generate trips and a joint staypoint/triplegs dataframe
+        stps, tpls, _ = ti.preprocessing.triplegs.generate_trips(stps, tpls,
+                                                                 gap_threshold=15,
+                                                                 id_offset=0)
+        assert pd.isna(stps['trip_id']).any()
+        assert pd.isna(stps['prev_trip_id']).any()
+        assert pd.isna(stps['next_trip_id']).any()
+        
```

Comment:
[isna](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.isna.html#pandas.DataFrame.isna) actually also accepts `None` values but I think this is OK